Adds phobia of heresy, instead of phobia of supernatural for heretic sacrifices.

Helg2 opened the pull request:
## About The Pull Request
Adds phobia of heresy, so after sacrificing someone they will only fear heretic stuff and not all magic.
## Why It's Good For The Game
Chaplains, wizards, cultists and heretics will be able to play after they get sacrificed.
## Changelog
:cl: Helg2
balance: Sacrificed people will now gain phobia of heresy, instead of supernatural, and heretics will be immune to it, so you'll be able to play as heretic normally.
/:cl:

Helg2 committed:
Yee

If heretic sac's heretic/wizard/cultist, they wont be scared because of themselves.

Helg2 committed:
Yee 1.5

I wonder how to make proper define, so it understands what role_wizard means

Helg2 committed:
Yee 2.0

ZephyrTFA commented on code/modules/antagonists/heretic/knowledge/sacrifice_knowledge/sacrifice_knowledge.dm:
```diff
@@ -370,6 +370,9 @@
 	sac_target.gain_trauma(/datum/brain_trauma/mild/phobia/supernatural, TRAUMA_RESILIENCE_MAGIC)
 
+	if(IS_WIZARD(sac_target) || IS_HERETIC(sac_target) || IS_CULTIST(sac_target))
+		sac_target.cure_trauma_type(/datum/brain_trauma/mild/phobia/supernatural,TRAUMA_RESILIENCE_MAGIC)
```
instead of applying and then removing the trauma, just don't apply it to begin with

Serxule commented:
what about chaplain? this leaves chaplain completely unable to do their job as even their jumpsuit triggers the trauma

Helg2 committed:
Yee 3.0

It works

Autisem commented:
> what about chaplain? this leaves chaplain completely unable to do their job as even their jumpsuit triggers the trauma

I feel since the chaplain is magic immune and set up as a "smiter of evil magic" he should also not get the trauma. makes sense IC wise atleast that they would not just stop believing in there religion if they get attacked by a heretic once

Autisem commented on code/modules/antagonists/heretic/knowledge/sacrifice_knowledge/sacrifice_knowledge.dm:
```diff
@@ -368,7 +368,10 @@
 	sac_target.clear_mood_event("shadow_realm")
-	sac_target.gain_trauma(/datum/brain_trauma/mild/phobia/supernatural, TRAUMA_RESILIENCE_MAGIC)
+
+	if(IS_WIZARD(sac_target) || IS_HERETIC(sac_target) || IS_CULTIST(sac_target) || sac_target.mind.holy_role)
```
```suggestion
	if(!(IS_WIZARD(sac_target) || IS_HERETIC(sac_target) || IS_CULTIST(sac_target) || sac_target.mind.holy_role))
		sac_target.gain_trauma(/datum/brain_trauma/mild/phobia/supernatural, TRAUMA_RESILIENCE_MAGIC)
```

Helg2 replied:
What does `!()` do?

Helg2 committed:
!()

Co-authored-by: Autisem <36102060+Autisem@users.noreply.github.com>

Helg2 committed:
What

Helg2 committed:
what

Jacquerel commented:
It’s make it less generally inconvenient but would it make sense to just replace the phobia with one of specifically heretic stuff (with a fuller list of items) and then not need to add exceptions for anything except sacrificing another heretic?  
That’d keep the intent of making it harder to fight heretics.

Helg2 commented:
Maybe you're right

Helg2 committed:
Herecy

Helg2 commented:
Why the fuck do i get errors from non-existing code
![image](https://user-images.githubusercontent.com/93882977/211199255-07bd5f43-c9e6-485c-9059-e2fcad2e65e8.png)

Jacquerel commented on code/_globalvars/phobias.dm:
```diff
@@ -378,6 +381,20 @@ GLOBAL_LIST_INIT(phobia_objs, list(
+		/obj/effect/visible_heretic_influence,
+		/obj/item/codex_cicatrix,
+		/obj/item/melee/rune_carver,
+		/obj/item/melee/sickly_blade
```
missing comma here

Helg2 committed:
non-existent coma

Helg2 committed:
Forgot those

Helg2 committed:
excessive

Jacquerel commented on code/_globalvars/phobias.dm:
```diff
@@ -378,6 +381,20 @@ GLOBAL_LIST_INIT(phobia_objs, list(
+		/obj/item/melee/sickly_blade,
+		/obj/item/toy/eldritch_book,
+		/obj/item/toy/reality_pierce,
+		/obj/structure/destructible/eldritch_crucible,
```
Suggestions for other items you could add:
`/obj/item/melee/touch_attack/mansus_fist` will get spooked if you see someone with a spooky hand active
`/obj/item/clothing/mask/madness_mask,` this one is _designed_ to upset people
`/obj/item/reagent_containers/cup/beaker/eldritch,` potion from the crucible
`/obj/item/clothing/suit/hooded/cultrobes/eldritch` if someone's wearing this they're going loud

Helg2 committed:
fix of something

Helg2 committed:
also those

Helg2 committed:
existent coma

Helg2 commented:
shit, its heresy and not herecy xd

Helg2 committed:
Heresy, not herecy

the-orange-cow commented:
> shit, its heresy and not herecy xd

Fear of hearsay imo

Profakos commented:
Definitely should have Hands in the string regex file. After all, the victim has spent quite a long time among them...

Helg2 commented:
> If not a phobia I'd still like for there to be some other setback (besides being out of it for two and a half minutes).
I think it'll be great bonus to heretic if he will drain other heretic's knowledge points, like 2 or 3 at max (to not ruin the game for other heretic)

Helg2 committed:
More to fear

Helg2 commented:
> Draining knowledge points seems a bit iffy, like it'd encourage people to spend it all as soon as they get them rather than hold them for options.

Just cancel the knowledge, if theres lack of them

Helg2 commented:
If someone wants to propose something to add to heresy phobia, go ahead. This pr will still take some time.

DaydreamIQ commented:
Heretics who get sacced could have maybe an increase to the cooldown of their spells? If such a thing is possible

Redrover1760 commented:
> If someone wants to propose something to add to heresy phobia, go ahead. This pr will still take some time.

I have an idea. If a Heretic is sacrificed, all of their spells (not rituals, as honestly, few people use them anyways and its mostly noncombat stuff) except Mansus grip are disabled as their connection to the Mansus has been weakened by their turmoil. The only way to regain access to their spells is to successfully sacrifice one of their targets. You could probably still get around this by trying to cooperate with a heretic trying to sac you by stashing the body ahead of time with them, but that risks you getting betrayed by said heretic fairly easily.

The only potential downside is that it may encourage stashing sacrifices targets instead of just saccing them, although that comes with the downside of not getting knowledge, but eh.

Alternatively, you could go the route of a ritual. Getting sacrificed grants you a ritual similar to the ritual of knowledge where you must collect 4 random items or whatever in order to use in order to re-activate your spells (and perhaps rituals, at this point)

Essentially, you are set back quite a bit by a really annoying progression hazard. Also, it goes without saying that you probably aren't allowed to learn any new knowledges in one of these states, although any existing knowledge points are simply locked away.

Helg2 committed:
Something like that

But it doesnt even work in build.bat so dunno
fixed some typos i found
i hope it'll show me errors though, because in build its not even linked to the part i added

SuperSlayer0 commented on code/modules/antagonists/heretic/heretic_knowledge.dm:
```diff
@@ -710,3 +710,19 @@
+
+/datum/heretic_knowledge/mansus_restoration_ritual/on_finished_recipe
+	var/datum/antagonist/heretic/our_heretic = IS_HERETIC(user)
+	ADD_TRAIT(our_heretic, TRAIT_ALLOW_HERETIC_CASTING, src)
```
Here you are just giving a heretic a permanent focus, without a need of wearing a focus item.

SuperSlayer0 commented on code/modules/antagonists/heretic/knowledge/sacrifice_knowledge/sacrifice_knowledge.dm:
```diff
@@ -368,7 +368,13 @@
+		sac_target.gain_trauma(/datum/brain_trauma/mild/phobia/heresy, TRAUMA_RESILIENCE_MAGIC)
+
+	if(IS_HERETIC(sac_target))
+		REMOVE_TRAIT(sac_target, TRAIT_ALLOW_HERETIC_CASTING, src)
```
Removing this trait wouldn't prevent the heretic from using spells. The heretic doesn't have it by default(so in some cases you are removing a trait which the heretic doesn't even have), and it is gained from wearing a focus item. So even if you will remove it, it can be easily gained back by dropping and putting back a focus item. Additionaly, you should do it like REMOVE_TRAIT(something, trait define, the trait source), because if you do it like you do you will try to remove a trait granted by the sacrifice ritual, which doesn't grant this trait.

Helg2 committed:
Merge branch 'master' into supernatural-fears-supernatural

theta891 commented:
> > If not a phobia I'd still like for there to be some other setback (besides being out of it for two and a half minutes).
> 
> I think it'll be great bonus to heretic if he will drain other heretic's knowledge points, like 2 or 3 at max (to not ruin the game for other heretic) Or rather dependant on total knowledge points.

maybe make it copy their learned knowledge so there is a benefit to saccing  another heretic but it doesnt entirely fuck them over

Helg2 committed:
yee

Helg2 committed:
yeet

Helg2 committed:
yee

Helg2 committed:
yee

Helg2 committed:
yee

Helg2 committed:
yee

minor typos change, what am i even doing

Helg2 committed:
yee

Profakos commented:
Hey can I remake and finish this for you? I want to see it in the game...

Helg2 commented:
> Hey can I remake and finish this for you? I want to see it in the game...

Sure, i dont mind
